### PR TITLE
refactor: get ROWKEY from key of Kafka record in topology steps

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 

Part of the primitive key work.

Switch the select mapper to get key columns from the key of the Kafka message

### Testing done 

`mvn test`

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java [hunk 1]
```diff
@@ -238,9 +239,13 @@ public String toString(final FormatOptions formatOptions) {
   }
 
   private Optional<Column> findColumnMatching(final Predicate<Column> predicate) {
+    // At the moment, it's possible for some column names to have multiple matches, e.g.
```
Comment (big-andy-coates) [inline ksql-common/src/main/java/io/confluent/ksql/schema/ksql/LogicalSchema.java]: Note: this is temporary and will go away once we're not copying key columns into the value and we ditch meta columns.

File: ksql-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenSpec.java [hunk 2]
```diff
@@ -137,46 +133,145 @@ CodeGenSpec build() {
...
+          ? resolveWindowed((Windowed<Struct>)key)
+          : resolveNonWindowed((Struct)key);
+    }
+
+    private Object resolveWindowed(final Windowed<Struct> windowedKey) {
+      final Window window = windowedKey.window();
+      final long start = window.start();
+      final String end = window instanceof SessionWindow ? String.valueOf(window.end()) : "-";
+      final Object key = resolveNonWindowed(windowedKey.key());
+      return String.format("%s : Window{start=%d end=%s}", key, start, end);
```
Comment (big-andy-coates) [inline ksql-execution/src/main/java/io/confluent/ksql/execution/codegen/CodeGenSpec.java]: This is a copy of what is currently done in `SourceBuilder`.   This is how it handles windowed keys. Yuck!  

Copy in `SourceBuilder` will go soon.

Will be fixed in time by structured keys work.

File: ksql-common/src/main/java/io/confluent/ksql/schema/ksql/Column.java [hunk 3]
```diff
@@ -29,9 +29,9 @@
 public final class Column implements SimpleColumn {
 
   public enum Namespace {
-    META,
     KEY,
-    VALUE
+    VALUE,
```
Comment (agavra) [inline ksql-common/src/main/java/io/confluent/ksql/schema/ksql/Column.java]: please add a comment explaining that the order of the enum is important (re `LogicalSchema`) - even though this is covered in tests it's nice to know that when reading the code

File: ksql-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java [hunk 4]
```diff
@@ -528,7 +527,7 @@ protected Object visitExpression(final Expression expression, final Void context
           );
 
       // we expect no column references, so we can pass in an empty generic row
```
Comment (agavra) [inline ksql-engine/src/main/java/io/confluent/ksql/engine/InsertValuesExecutor.java]:
```suggestion
      // we expect no column references, so we can pass in an empty generic key and row
```

File: ksql-engine/src/test/java/io/confluent/ksql/codegen/CodeGenRunnerTest.java [hunk 5]
```diff
@@ -875,7 +923,7 @@ public void shouldChoseFunctionWithCorrectNumberOfArgsWhenNullArgument() {
 
         return analysis.getSelectExpressions().stream()
             .map(exp -> codeGenRunner.buildCodeGenFromParseTree(exp.getExpression(), "Select"))
-            .map(md -> md.evaluate(input))
+            .map(md -> md.evaluate(StructKeyUtil.asStructKey("rowKey"), input))
```
Comment (agavra) [inline ksql-engine/src/test/java/io/confluent/ksql/codegen/CodeGenRunnerTest.java]:
```suggestion
            .map(md -> md.evaluate(key(), input))
```

File: ksql-common/src/main/java/io/confluent/ksql/schema/ksql/Column.java (hunk 3 above)
Comment (big-andy-coates) [reply ksql-common/src/main/java/io/confluent/ksql/schema/ksql/Column.java]: It won't be needed for long, but OK

**Pull request merged**